packages/gestalt/src/Upsell.js

```diff
@@ -154,15 +157,21 @@ export default function Upsell({
             </Box>
           </Box>
         </Box>
-        <Box smDisplay="flex" marginStart="auto" smMarginEnd={4} smPaddingY={3}>
-          {secondaryAction && responsiveMinWidth !== 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} />
-          )}
-          {primaryAction && <UpsellAction type="primary" data={primaryAction} />}
-          {secondaryAction && responsiveMinWidth === 'xs' && (
-            <UpsellAction type="secondary" data={secondaryAction} stacked={!!secondaryAction} />
-          )}
-        </Box>
+        {children ? (
```

Comment:
Hm…this is a bit of a challenge. We need to convey to the user that they can provide children or primary/secondaryActions, but not both. How do you think we should handle that?

Reply:
Is this a case where thorough documentation is enough?

I feel like we had a similar case with Modal where there was a certain combination of props you could have (like `x` or `y and z` but not `x and y and z`). I think @AlbertCarreras and I tried using flow to type it so that it would only allow the right combinations but if I remember correctly that caused issues when we went to pinboard and had to be removed.

Reply:
Yep, there's not a good way to prevent this with Flow that I'm aware of. I think more likely we'll need to add a lint rule to guard against invalid prop combinations. That can be a follow-on PR though…if you're also interested in working on that, check out some of the existing lint rules in `packages/eslint-plugin-gestalt/src` for inspiration 🙂

Reply:
Sweet thanks! I'll address that in a follow up! I tried adding it in this PR but it looks like its going to take me a little bit more time to make sure I'm properly finding the children of Upsell when a form is being used.